api/src/wfl/module/covid.clj

```diff
@@ -180,25 +180,135 @@
+   from the `_tdr-job-metadata` map, update timestamp with real now."
+  [{:keys [details] :as _source}
+   {:keys [id job_status snapshot_id] :as _tdr-job-metadata}]
+  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+    (jdbc/update! tx details {:snapshot_creation_job_status job_status
+                              :snapshot_id snapshot_id
+                              :updated  (OffsetDateTime/now)}
+      ;; we don't have the primary key reference of details table here,
+      ;; so use job-id for where clause, since job-id is unique in TDR,
+      ;; this shall be fine
+                  ["snapshot_creation_job_id = ?" id])))
+
+(defn ^:private write-snapshots-creation-jobs
+  "Write all `snapshots-creation-jobs` along with the `shards` they
+   try to snapshot into source `details` table, with the frozen `now`.
+   Also initialize all jobs statuses to running."
+  [{:keys [last_checked details] :as _source} now shards snapshots-creation-jobs]
+  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+    (jdbc/insert-multi! tx
+                        details
+                        (mapv (fn [shard id] {:snapshot_creation_job_id id
+                                              :snapshot_creation_job_status "running"
+                                              :datarepo_row_ids         (vec shard)
+                                              :start_time               last_checked
+                                              :end_time                 now})
+                              shards snapshots-creation-jobs))))
+
+(defn ^:private update-last-checked
+  "Update the `last_checked` field in source table with
+   the frozen `now`."
+  [{:keys [id] :as _source} now]
+  (jdbc/with-db-transaction [tx (postgres/wfl-db-config)]
+    (jdbc/update! tx tdr-source-table {:last_checked now} ["id = ?" id])))
+
 ;; Create and add new snapshots to the snapshot queue
-(defn ^:private update-tdr-source [source]
-  (letfn [(find-new-rows       [source now] [])
-          (make-snapshots      [source row-ids])
-          (write-snapshots     [source snapshots])
-          (update-last-checked [source now])]
-    (let [now     (OffsetDateTime/now)
-          row-ids (find-new-rows source now)]
-      (when-not (empty? row-ids)
-        (write-snapshots source (make-snapshots source row-ids)))
-      (update-last-checked source now))))
+(defn ^:private update-tdr-source
+  "Check for new data in TDR, create new snapshots, insert
+   resulting job creation ids into database and update the
+   timestamp for next time using transaction TX."
+  [source]
+  ;; attempt to snapshot new rows in TDR
+  (let [utc-now-obj           (OffsetDateTime/now (ZoneId/of "UTC"))
+        utc-now               (.format utc-now-obj (DateTimeFormatter/ofPattern "yyyy-MM-dd HH:mm:ss"))
+        row-ids               (find-new-rows source utc-now)
+        [shards tdr-job-ids]  (create-snapshots source utc-now-obj row-ids)]
```

Comment:
Can you compose these three into something like the following as these temporary names aren't used anywhere else

```clojure
(defn ^:private find-and-snapshot-new-rows [souce utc-now]
  (let [date-format (DateTimeFormatter/ofPattern "yyyy-MM-dd HH:mm:ss")]
    (->> (.format utc-now date-format)
         (find-new-rows source)
         (create-snapshots souce utc-now))))
```

Reply:
Also, if you're using this for big-query, don't you mean "yyyy-MM-ddTHH:mm:ss"

Reply:
https://console.cloud.google.com/bigquery?authuser=1&project=wfl-dev&ws=!1m5!1m4!4m3!1stdr-prod-covid19-viralseq!2sdatarepo_CDC_COVID19_Surveillance_v1!3sflowcells!1m0&j=bq:us-central1:bquxjob_69e0cf3b_1797b6d1bfa&page=queryresults

Reply:
updated